### PR TITLE
Rails Recipe App

### PR DESCRIPTION
**Pull request by Chimoneg27**
### For this project me and @nilab-khwaja did the following 

- Use Ruby gems as a software packages system.
- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Understand Rails naming conventions.
- Use params from browser requests in a safe way.
- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a web app that requires the user to log in.
- Describe the difference between authorization and authentication.
- Use Devise gem for authentication.
- Understand how sessions and cookies can support authentication.
- Limit access to web app resources based on authorization rules.
- Write unit tests using Rspec
- Write integration tests with capybara gem

### CODE REVIEW

File: spec/Features/food_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+RSpec.describe 'Food Index Page' do
+  before(:each) do
+    @user = create(:user)
+    @user.confirm
+    @food = create(:food, user_id: @user.id)
+    login_as(@user, scope: :user)
+  end
+
+  it 'can see the add food button' do
```
Comment (1GORDON) [inline spec/Features/food_index_spec.rb]: Great job team 🤝
- You have done a great job implementing your tests. However, some of your test cases are still failing the checks. Kindly fix these tests and ensure all test cases are passing.
fail-test

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Chimoneg27, and @nilab-khwaja ✋

Good job so far!

## Highlights!
- Has a navigation menu ✔️ 
- Uses Devise ✔️
- Uses Capybara and RSpec testing libraries ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
- Kindly remove commented code from your `routes.rb` file. Leaving commented code in your project is regarded as a bad practice.
comments

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Chimoneg27** [comment]
**Note**

Dear reviewer the Mac and cheese recipe was the first recipe created before authorization and authentication, so Just ignore it

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  resources :recipes, only: [:index, :show, :new, :create, :destroy] do
+    resources :recipe_foods, only: [:new, :create, :edit, :update, :destroy]
+      member do
+              put :toggle_public
+    end
+  end
+  resources :users, only: [:index, :show, :new, :create, :destroy]
+  # root to: 'foods#index'
+  get '/public_recipes', to: 'recipes#public_recipes', as: 'public_recipes'
+  # get "up" => "rails/health#show", as: :rails_health_check
```
Comment (danmainah) [inline config/routes.rb]: - [ ] @Team Great work on your app, however, you forgot to implement the change requested to remove the commented code from your routes file, its bad practice having commented code in your codebase.

File: spec/Features/food_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+RSpec.describe 'Food Index Page' do
+  before(:each) do
+    @user = create(:user)
+    @user.confirm
+    login_as(@user, scope: :user)
+    @food = create(:food, user_id: @user.id)
+  end
+
+  it 'can see the add food button' do
```
Comment (danmainah) [inline spec/Features/food_index_spec.rb]: - [ ] @Team Great work trying to fix the test errors, however some of your tests are still failing, kindly consider fixing these tests to ensure all tests are passing. 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Chimoneg27 and @nilab-khwaja , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Changed the tests setup to try and fix tests that are failing.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by danmainah** [comment]
> **Note**
> 
> Dear reviewer the Mac and cheese recipe was the first recipe created before authorization and authentication, so Just ignore it

@Chimoneg27 Sorry for the late reply, since these are part of the codebase they will be tested, you can fix their error or remove them if they are not part of the project requirement.

File: spec/Features/food_index_spec.rb (hunk 3 above)
Comment (Chimoneg27) [reply spec/Features/food_index_spec.rb]: @danmainah they pass on my pc

Comment (Chimoneg27) [reply spec/Features/food_index_spec.rb]: @danmainah they pass on my pc

**Comment by Chimoneg27** [comment]
**Note**

Dear reviewer the folder for testing the feature has an issue on github. It works when its called features instead of Features

**Review by prolajumokeoni — APPROVED** [review]
Hi @Chimoneg27 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Tests are passing  ✅ 
  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**